src/components/TodoList/TodoList.tsx

```diff
@@ -1,100 +1,111 @@
-import React from 'react';
+import { FC } from 'react';
+import { Filter } from '../../helpers/constants';
+import { Todo } from '../../types/Todo';
 
-export const TodoList: React.FC = () => (
-  <table className="table is-narrow is-fullwidth">
-    <thead>
-      <tr>
-        <th>#</th>
-        <th>
-          <span className="icon">
-            <i className="fas fa-check" />
-          </span>
-        </th>
-        <th>Title</th>
-        <th> </th>
-      </tr>
-    </thead>
+type Props = {
+  todos: Todo[],
+  filter: string,
+  selectTodo: (todoId: number) => void,
+  selectedTodo: number,
+  search: string,
+};
 
-    <tbody>
-      <tr data-cy="todo" className="">
-        <td className="is-vcentered">1</td>
-        <td className="is-vcentered" />
-        <td className="is-vcentered is-expanded">
-          <p className="has-text-danger">delectus aut autem</p>
-        </td>
-        <td className="has-text-right is-vcentered">
-          <button data-cy="selectButton" className="button" type="button">
-            <span className="icon">
-              <i className="far fa-eye" />
-            </span>
-          </button>
-        </td>
-      </tr>
-      <tr data-cy="todo" className="has-background-info-light">
-        <td className="is-vcentered">2</td>
-        <td className="is-vcentered" />
-        <td className="is-vcentered is-expanded">
-          <p className="has-text-danger">quis ut nam facilis et officia qui</p>
-        </td>
-        <td className="has-text-right is-vcentered">
-          <button data-cy="selectButton" className="button" type="button">
-            <span className="icon">
-              <i className="far fa-eye-slash" />
-            </span>
-          </button>
-        </td>
-      </tr>
+export const TodoList: FC<Props> = ({
+  todos,
+  filter,
+  selectTodo,
+  selectedTodo: selectedTodoId,
+  search,
+}) => {
+  const todosForRender = todos
+    .filter((todo: Todo) => {
+      switch (filter) {
+        case Filter.all:
+          return todo;
+        case Filter.completed:
+          return todo.completed;
+        case Filter.active:
+          return !todo.completed;
+        default:
+          throw new Error('Error during filtering');
+      }
+    })
```

Comment:
do different filters inside every case. because for all filter it would be extra loop